src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -27,29 +40,329 @@ const ActorInit En_Door_InitVars = {
+};
+
+Gfx* D_809FCEE4[5][2] = {
+    { 0x0400ECB8, 0x0400EE00 },
+    { 0x0600F998, 0x0600F938 },
+    { 0x06004958, 0x06004A10 },
+    { 0x060013B8, 0x06001420 },
+    { 0x050047A0, 0x05004978 },
+};
+
+extern Gfx D_0400ECB8[];
+extern Gfx D_0400EE00[];
+
+extern AnimationHeader D_0400E758;
+extern SkeletonHeader D_0400FF78;
+
+#ifdef NON_MATCHING
+// Regalloc only
+void EnDoor_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad1;
+    EnDoorInfo* objectInfo;
+    s32 i;
+    s32 objIndex;
+    f32 xOffset;
+    f32 zOffset;
+    EnDoor* this = THIS;
+
+    objectInfo = &D_809FCEA0[0];
+    Actor_ProcessInitChain(&this->actor, D_809FCEC4);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_0400FF78, &D_0400E758, this->limbDrawTable, this->transitionDrawTable, 5);
+    for (i = 0; i < ARRAY_COUNT(D_809FCEA0) - 2; i++, objectInfo++) {
+        if (globalCtx->sceneNum == objectInfo->sceneNum) {
+            break;
+        }
+    }
+    if (i >= ARRAY_COUNT(D_809FCEA0) - 2 && Object_GetIndex(&globalCtx->objectCtx, OBJECT_GAMEPLAY_FIELD_KEEP) >= 0) {
+        objectInfo++;
+        if (!globalCtx->sceneNum) {}
+    }
+
+    this->displaylistIdx = objectInfo->displaylistIdx;
+    objIndex = Object_GetIndex(&globalCtx->objectCtx, objectInfo->objectId);
+    if (objIndex < 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->currentObject = objIndex;
+    this->displaylistIdx = objectInfo->displaylistIdx;
+    if (this->currentObject == this->actor.objBankIndex) {
+        EnDoor_SetupType(this, globalCtx);
+    } else {
+        this->actionFunc = EnDoor_SetupType;
+    }
+
+    // Double doors
+    if (this->actor.params & 0x40) {
+        EnDoor* attached;
+
+        xOffset = Math_Coss(this->actor.shape.rot.y) * 30.0f;
+        zOffset = Math_Sins(this->actor.shape.rot.y) * 30.0f;
+        attached = (EnDoor*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_DOOR, 
+                    this->actor.posRot.pos.x + xOffset, this->actor.posRot.pos.y, this->actor.posRot.pos.z - zOffset, 
+                    0, this->actor.shape.rot.y + 0x8000, 0, this->actor.params & ~0x40);
+        if (attached != NULL) {
+            attached->unk_192 = 1;
+        }
+        this->actor.posRot.pos.x -= xOffset;
+        this->actor.posRot.pos.z += zOffset;
+    }
+    Actor_SetHeight(&this->actor, 70.0f);
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/EnDoor_Init.s")
+#endif
+
+void EnDoor_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    TransitionActorEntry* transitionEntry;
+    EnDoor* this = THIS;
+
+    transitionEntry = &globalCtx->transitionActorList[(u16)this->actor.params >> 0xA];
+    if (transitionEntry->id < 0) {
+        transitionEntry->id = -transitionEntry->id;
+    }
+}
+
+void EnDoor_SetupType(EnDoor* this, GlobalContext* globalCtx) {
+    s32 doorType;
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->currentObject)) {
```

Comment:
maybe `requiredObject`(`Id`) is better than currentObject, because the "true" current object is `Actor.objectBankIndex` while `EnDoor.currentObject` is the object that should be actually used